activerecord/lib/active_record/migration.rb

```diff
@@ -967,7 +967,7 @@ def schema_migrations_table_name
       end
 
       def get_all_versions(connection = Base.connection)
-        if connection.table_exists?(schema_migrations_table_name)
+        if connection.data_source_exists?(schema_migrations_table_name)
```

Comment:
It seems that `schema_migrations_table_name` table is a table, not a view. Should we use `table_exists?`?